### PR TITLE
Allow importing Svg icons

### PR DESCRIPTION
**Pull request by alex-ds13**
This PR enables the addition of custom icons to the font.

To do so we simply need to put the svg image files on the same folder as the TOML file and then on the `[glyphs]` section we set the name we want for said icon as the key and the value needs to be the file name for the svg image that will be imported for that icon.

This file name **needs** to end in ".svg" so that we recognize the entry as a custom svg icon that needs to be imported.

As an example if we wanted an `iced` icon, we simply need to add an svg image to the folder, let's say it's named `iced.svg` then on the `example-icons.toml` file we would add it like this:
```toml
module = "icon"

[glyphs]
edit = "fontawesome-pencil"
save = "entypo-floppy"
trash = "typicons-trash"
iced = "iced.svg"
```

### **Note:**
It's important that the svg file is correctly structured to be able to be imported and some svg tags will be ignored. Only the `path` will be considered, any `rect`, `circle` and others like it will simply be ignored.
Basically you can read this from fontello wiki to know more about this: https://github.com/fontello/fontello/wiki/How-to-use-custom-images#importing-svg-images

In case you are facing any issue with an svg file, first try importing it on the fontello site. If it doesn't work there either than you'll know that fontello doesn't support it and it needs to be changed like mentioned on the link above.

### CODE REVIEW

**Comment by alex-ds13** [comment]
We probably can remove that "wrong" `iced_logo` icon.

Also if you want I can remove the changes I've made to the `example/src/main.rs`. I've made them mostly to test that the icons were being properly imported.

**Comment by alex-ds13** [comment]
I've also just noticed an issue now. I've added this line to the `build.rs` file:
```rust
    println!("cargo::rerun-if-changed=fonts/*.svg");
```
Thinking this would trigger a build whenever we changed an svg file, because previously if I only made a change on an svg file it wouldn't trigger a new build. And the line above sort of worked, as in it did started building whenever I've made a change to an svg file. However I've just noticed that now it is making a new `iced_fontello` build even when I make a change to the `example/src/main.rs` file which it shouldn't!

If I remove that line it stops doing it... However I think it is important to trigger a build whenever a change is made to an svg file!

Do you know any workaround for this? I've just learned about this `cargo::rerun-if-changed` a couple of days ago when I saw it on `iced_fontello`! 😄

**Comment by alex-ds13** [comment]
@hecrj I've finally got the time to refactor one of my projects to use `iced_fontello` with this PR and discovered that there was a bug where when using custom-icons it was always creating a different hash on rebuild. This is now fixed on the last commit I've just pushed. From using this on my own project this PR's version seems to be working quite well!

I'm posting this just in case you want to revisit the idea of merging this.

File: src/svg_parser.rs [hunk 1]
```diff
@@ -0,0 +1,153 @@
+use std::{collections::BTreeMap, path::Path};
+
+use svg::{
+    node::element::tag::{self, Type},
+    parser::Event,
+};
+use svg_path_ops::pt::PathTransformer;
```
Comment (hecrj) [inline src/svg_parser.rs:7]: Any chance `usvg` may make this easier for us here?

Comment (hecrj) [reply src/svg_parser.rs:7]: I was wondering if `usvg` already simplified these paths for us.

Comment (alex-ds13) [reply src/svg_parser.rs]: > I was wondering if `usvg` already simplified these paths for us.

So I've finally had some time to look into this.

The first time I looked at `usvg` I didn't see that it used the `tiny_skia_path` internally for the actual paths and paths operations. Knowing this, I'm able to use the paths from `usvg` and apply it's transforms. It has the added benefit that the function `path.abs_transform()` gets the absolute transformations for a path with all the transforms from it's parents included. Also `usvg` makes all paths absolute so turning the path into a string is simpler. *(I couldn't find a way to turn the final path into a string, except for the `Debug` impl for the `tiny_skia_path::Path`, so I've copied that to the `write_path` function)*

In the end we can also apply a translate and scale transform to the final path like we were doing before to make the glyph have the size that fontello expects and be on the correct position.

So I guess in the end this did simplify things quite a bit as the code seems simpler to follow and understand now.

Let me know what you think!
